backend/controllers/playlist.controller.js

```diff
@@ -0,0 +1,130 @@
+                }
+                else {
+                    return { "_id": p._id, "name": p.name, "videos": [...p.videos, videoID] }
+                }
+            }
+            return p
+        })
+
+        const updateDetails = {
+            user: id,
+            playlists: updateItems
+        }
+        const final = await Playlist.findOneAndUpdate({ _id: lists._id }, { $set: updateDetails }, { new: true })
+        res.status(200).send(final)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+//@route   POST api/playlist
+//@desc    Add New Playlist
+//@access  Private
+const addNewPlaylist = async (req, res) => {
+    try {
+        const id = req.user
+        const { name, videoID } = req.body
+        let pList = await Playlist.findOne({ user: id })
+        const present = pList.playlists.some((p) => p.name === name)
+        if (present) {
+            return res.status(400).json({ success: false, message: 'Playlist with this name already present' })
+        }
+        const ToAdd = {
+            name: name,
+            videos: [videoID]
+        }
+        pList.playlists = _.concat(pList.playlists, ToAdd)
+        await pList.save()
+        res.status(200).send(pList)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+
+//@route   DELETE api/playlist/:playlistID
+//@desc    Delete Playlist
+//@access  Private
+const deletePlaylist = async (req, res) => {
+    try {
+        const id = req.user;
+        let lists = await Playlist.findOne({ user: id })
+        const { playlistID } = req.params
+        const present = lists.playlists.some((p) => p._id == playlistID)
+        if (present) {
+            const filterPlaylist = lists.playlists.filter((p) => p._id != playlistID)
+            lists.playlists = filterPlaylist
+            await lists.save()
+            return res.status(200).send(lists)
+        }
+        return res.status(400).json({ success: false, message: 'No such Playlist exists' })
+
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+//@route   DELETE api/playlist/:playlistID/:videoID
+//@desc    Delete Playlist
+//@access  Private
+
+const removeVideoFromPlaylist = async (req, res) => {
+    try {
+        const userID = req.user
+        const { playlistID, videoID } = req.params
+        const requiredPlaylist = await Playlist.findOne({ user: userID })
+        const finalVideos = requiredPlaylist.playlists.map((one) => {
+            if (one._id == playlistID) {
+                const findVideo = one.videos.some((v) => v == videoID)
+                if (findVideo) {
+                    const filtervideo = one.videos.filter((video) => video !== videoID)
+                    return { "_id": one._id, "name": one.name, "videos": filtervideo }
+                }
+                return one
```

Comment:
You could improve on the namings for such variables. In this case, it could be renamed to `playlist` or `list` or `singleList`...

Reply:
done :)